File: Rust/distance_calc.rs
fn distance2d(from: [f64; 2], to: [f64; 2]) -> f64 {
	let diff_x = (from[0] - to[0]).abs();
	let diff_y = (from[1] - to[1]).abs();
	let delta = (diff_x.powi(2) + diff_y.powi(2)).sqrt();
	return delta;
}

fn distance3d(from: [f64; 3], to: [f64; 3]) -> f64 {
	let from2d = [from[0], from[2]];
	let to2d = [to[0], to[2]];
	let diff_x = distance2d(from2d, to2d);
	let diff_y = (from[1] - to[1]).abs();
	let delta = (diff_x.powi(2) + diff_y.powi(2)).sqrt();
	return delta;
}

fn main() {
	let a = [1.0, 2.0, 3.0];
	let b = [4.0, 5.0, 6.0];
	println!("{}", distance3d(a, b));
}

File: C#/DistanceCalc.cs
using System;

// by iRRiNiS
	
public class Program
{
	public static double Distance2D(double[] From, double[] To)
	{
	    double DiffX = Math.Abs(From[0] - To[0]);
		double DiffY = Math.Abs(From[1] - To[1]);
		double Delta = Math.Sqrt(Math.Pow(DiffX, 2.0) + Math.Pow(DiffY, 2.0));
		return Delta;
	}
	
	public static double Distance3D(double[] From, double[] To)
    {
		double[] From2D = {From[0], From[2]};
		double[] To2D = {To[0], To[2]};
		double DiffX = Distance2D(From2D, To2D);
		double DiffY = Math.Abs(From[1] - To[1]);
		double Delta = Math.Sqrt(Math.Pow(DiffX, 2.0) + Math.Pow(DiffY, 2.0));
		return Delta;
	}

	public static void Main()
	{
		double[] A = {1.0, 2.0, 3.0};
		double[] B = {4.0, 5.0, 6.0};
		
		Console.WriteLine(Distance3D(A, B));
	}
}
